p2p/src/net/mock/backend.rs

```diff
@@ -319,6 +333,10 @@ where
             .map_err(P2pError::from)
     }
 
+    async fn handle_announcement(&mut self, _announcement: Announcement) -> crate::Result<()> {
+        todo!();
```

Comment:
?

Reply:
The block announcement (pubsub) wasn't implemented for the mock backed, so in this pull request I have added the minimal required parts in order to pass the existing tests. I'm going to create an issue to finish this implementation.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/488